client/network/src/service.rs

```diff
@@ -377,6 +378,16 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 			_marker: PhantomData,
 		});
 
+		let event_streams = out_events::OutChannels::new(metrics.as_ref().map(|metrics| {
```

Comment:
I would prefer to pass a Prometheus registry down to `OutChannels::new` instead of passing multiple metrics down which also need to be registered here.

I think handling metric definition and registration in `out_events` reduces the noise within this upper module.